### PR TITLE
Add flow for  Running tests with latest version of DDC Smart Contract

### PR DESCRIPTION
**Pull request by tabakopavel**

### CODE REVIEW

File: .github/workflows/run-tests-on-new-version-of-ddc-sc.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+      - name: Run tests with latest version of smart contract
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          DDC_SMART_CONTRACT_VERSION: ${{ github.event.client_payload.DDC_SMART_CONTRACT_VERSION }}
+        run: |
+          echo "DDC_SMART_CONTRACT_VERSION::$DDC_SMART_CONTRACT_VERSION"
+          docker build --build-arg ECR_REGISTRY=$ECR_REGISTRY --build-arg DDC_SMART_CONTRACT_VERSION=$DDC_SMART_CONTRACT_VERSION -f Dockerfile.tests -t pos-network-node:test .
+      - name: Free space
```
Comment (alexremn) [inline .github/workflows/run-tests-on-new-version-of-ddc-sc.yml:38]: Why do we need to know free space on managed build agent in the end?

File: .github/workflows/run-tests-on-new-version-of-ddc-sc.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        uses: easimon/maximize-build-space@v3
+        with:
+          root-reserve-mb: 512
+          swap-size-mb: 1024
+          remove-dotnet: 'true'
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Checkout submodules
+        run: git submodule update --init --recursive
+      - name: Free space
```
Comment (alexremn) [inline .github/workflows/run-tests-on-new-version-of-ddc-sc.yml:20]: Why do we need to check the free space before build?

File: .github/workflows/run-tests-on-new-version-of-ddc-sc.yml (hunk 1 above)
Comment (tabakopavel) [reply .github/workflows/run-tests-on-new-version-of-ddc-sc.yml:38]: I don't know, it is already in other workflows. If somebody add it, i think it is really needed.

File: .github/workflows/run-tests-on-new-version-of-ddc-sc.yml (hunk 2 above)
Comment (tabakopavel) [reply .github/workflows/run-tests-on-new-version-of-ddc-sc.yml:20]: I don't know, it is already in other workflows. If somebody add it, i think it is really needed.

**Review by alexremn — APPROVED** [review]

**Pull request merged**